controlplane/kubeadm/controllers/controller.go

```diff
@@ -283,6 +283,14 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	// source ref (reason@machine/name) so the problem can be easily tracked down to its source machine.
 	conditions.SetAggregate(controlPlane.KCP, controlplanev1.MachinesReadyCondition, ownedMachines.ConditionGetters(), conditions.AddSourceRef())
 
+	if err := r.remediateUnhealthy(ctx, logger, controlPlane); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); err != nil {
```

Comment:
I think we should add a comment here explaining that this call ensures that the subsequent code is executed only if
- nothing is currently being deleted
- control-plane stati pods are in a healthy state
- etcd is in healthy state

(we know this by heart now, but this might be not so obvious to other people reading the code base)

Reply:
i don't like comments like this, because if someone changes the implementation of `reconcileHealth` they will no longer be true.

Reply:
I'll add this

Reply:
What if we just move this code to outside `reconcileHealth`? It seems like it doesn't really belong in it

```
    if controlPlane.HasDeletingMachine() {
      return &capierrors.RequeueAfterError{RequeueAfter: deleteRequeueAfter}
    }
```

Reply:
I'm ok with this if we add a comment to that explains why we are requeing when we are deleting.
Also, is it necessary requeue after of a new reconciliation will be triggered when one of the underlying machines changes (or gets deleted)